src/libYARP_OS/include/yarp/os/impl/BufferedConnectionWriter.h

```diff
@@ -364,7 +364,6 @@ class YARP_OS_impl_API yarp::os::impl::BufferedConnectionWriter : public Connect
             appendStringBase(str);
         } else {
             ConstString s = str;
-            str += terminate;
```

Comment:
This fix seems wrong, I think that who wrote this meant to write `s += terminate` instead of `str += terminate`